### ISSUE TITLE
macOS Audio Device Buffer Underrun Detected

### ISSUE THREAD

**Issue by denppa** [mpv-player/mpv#13189]
### Important Information

Provide following Information:
- mpv Version:
```bash
mpv 0.37.0 Copyright © 2000-2023 mpv/MPlayer/mplayer2 projects
 built on Nov 21 2023 18:47:46
libplacebo version: v6.338.1
FFmpeg version: 6.1
FFmpeg library versions:
   libavutil       58.2.100 (runtime 58.29.100)
   libavcodec      60.3.100 (runtime 60.31.102)
   libavformat     60.3.100 (runtime 60.16.100)
   libswscale      7.1.100 (runtime 7.5.100)
   libavfilter     9.3.100 (runtime 9.12.100)
   libswresample   4.10.100 (runtime 4.12.100)
```
- macOS Version:
`macOS Sonoma 14.0`
- Source of the mpv binary or bundle:
- `homebrew`
- If known which version of mpv introduced the problem
Not sure.
- hardware:
M1 mac mini, ram at ~20%, CPU around ~20%. External audio dac connected via USB as output device

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Playing a file with the below video and audio format without any options create an audio device buffer underrun. Adding `--audio-buffer=1` helps out significantly.

### Expected behavior
Play without dropped audio frames.
### Actual behavior
Drops audio frames due to buffer underrun, causing glitches in playback.
### Log file
```bash
Resuming playback. This behavior can be disabled with --no-resume-playback.
 (+) Video --vid=1 (*) (vp9 1280x720 29.970fps)
 (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
AO: [coreaudio] 48000Hz stereo 2ch floatp
VO: [libmpv] 1280x720 yuv420p
Audio device underrun detected.
Audio device underrun detected.
Saving state.
Exiting... (Quit)
```

### Sample files

The file is downloaded via yt-dlp on youtube.

**Comment by quietvoid** [comment]
FWIW I play same Opus 2.0 audio from YouTube just fine with default settings, no such errors here.
System: M2 MBP

**Comment by rgaufman** [comment]
Any solution other than switching from gpu from gpu-next? - experiencing stuttering on a Macbook Pro 14" M3 Max.

**Comment by ruihe774** [comment]
@Akemi Is this the related to #7172? I wonder whether you've found out the cause. If it is still unclear maybe we can simply increase the `device_buffer` as a workaround.

**Comment by rgaufman** [comment]
This is happening for me with both vo=gpu and vo=gpu-next on the M3 Max. I'll try with audio-buffer=1.

**Comment by Akemi** [comment]
fixed by #14208
